frontend/App.js

```diff
@@ -11,16 +11,25 @@ import {
   TouchableOpacity,
   View,
 } from 'react-native';
-import Exponent, { Constants, ImagePicker, registerRootComponent } from 'expo';
+import Exponent, { Constants, ImagePicker, Permissions, registerRootComponent } from 'expo';
 
-export default class App extends React.Component {
-  state = {
-    image: null,
-    uploading: false,
-  };
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      image: null,
+      uploading: false,
+    };
+
+    // bind(s)
+    this.allowPermission = this.allowPermission.bind(this);
```

Comment:
arrow function

Reply:
After the arrow function is added,  `this.state = {}` should move to the outer scope. `constructor` function should be deleted.